io/PureC_IO_Bigarray.re

```diff
@@ -0,0 +1,69 @@
+open Odiff.ImageIO;
+
+module IO: Odiff.ImageIO.ImageIO = {
+  type rowPointers = int;
+  type t = {
+    rowPointers,
+    bigarray:
+      Bigarray.Array1.t(int, Bigarray.int8_unsigned_elt, Bigarray.c_layout),
```

Comment:
One thing we could think about, is to make the Bigarray use an `uint32` to store the complete pixel instead of a `uint8` for each color channel. That way reading the pixel could possibly be faster, as we only need to read once from the array and can compute the different channels from that one value.